src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeMemory.Unix.cs

```diff
@@ -0,0 +1,212 @@
+        ///     <para>This method is a platform dependent aligned reallocation API such as <c>_aligned_realloc</c> on Win32.</para>
+        ///     <para>This method is not compatible with <see cref="Free" /> or <see cref="Realloc" />, instead <see cref="AlignedFree" /> or <see cref="AlignedRealloc" /> should be called.</para>
+        /// </remarks>
+        [CLSCompliant(false)]
+        public static void* AlignedRealloc(void* ptr, nuint byteCount, nuint alignment)
+        {
+            void* newPtr = AlignedAlloc(byteCount, alignment);
+            Buffer.Memmove(ref *(byte*)newPtr, ref *(byte*)ptr, byteCount);
+
+            AlignedFree(ptr);
+            return newPtr;
+        }
+
+        /// <summary>Allocates a block of memory of the specified size, in bytes.</summary>
+        /// <param name="byteCount">The size, in bytes, of the block to allocate.</param>
+        /// <returns>A pointer to the allocated block of memory.</returns>
+        /// <exception cref="OutOfMemoryException">Allocating <paramref name="byteCount" /> of memory failed.</exception>
+        /// <remarks>
+        ///     <para>This method allows <paramref name="byteCount" /> to be <c>0</c> and will return a valid pointer that should not be dereferenced and that should be passed to free to avoid memory leaks.</para>
+        ///     <para>This method is a thin wrapper over the C <c>malloc</c> API.</para>
+        /// </remarks>
+        [CLSCompliant(false)]
+        public static void* Alloc(nuint byteCount)
+        {
+            // The C standard does not define what happens when size == 0, we want an "empty" allocation
+            void* result = Interop.Sys.Malloc((byteCount != 0) ? byteCount : 1);
+
+            if (result == null)
+            {
+                ThrowHelper.ThrowOutOfMemoryException();
+            }
+
+            return result;
+        }
+
+        /// <summary>Allocates a block of memory of the specified size, in elements.</summary>
+        /// <param name="elementCount">The count, in elements, of the block to allocate.</param>
+        /// <param name="elementSize">The size, in bytes, of each element in the allocation.</param>
+        /// <returns>A pointer to the allocated block of memory.</returns>
+        /// <exception cref="OutOfMemoryException">Allocating <paramref name="elementCount" /> * <paramref name="elementSize" /> bytes of memory failed.</exception>
+        /// <remarks>
+        ///     <para>This method allows <paramref name="elementCount" /> and/or <paramref name="elementSize" /> to be <c>0</c> and will return a valid pointer that should not be dereferenced and that should be passed to free to avoid memory leaks.</para>
+        ///     <para>This method is a thin wrapper over the C <c>malloc</c> API.</para>
+        /// </remarks>
+        [CLSCompliant(false)]
+        public static void* Alloc(nuint elementCount, nuint elementSize)
```

Comment:
`Alloc(nuint elementCount, nuint elementSize)` and `AllocZeroed(nuint byteCount)` are identical between Windows and Unix. They can be in the share file too.

Reply:
Fixed.